src/jit/lir.cpp

```diff
@@ -581,6 +649,9 @@ void LIR::Range::InsertBefore(const Range& range, GenTree* insertionPoint)
     assert(!range.IsEmpty());
     assert(!range.IsSubRange());
 
+    assert(range.Begin()->gtPrev == nullptr);
+    assert(range.End()->gtNext == nullptr);
```

Comment:
The `!range.IsSubRange()` check above should cover these asserts. Same for those added to `InsertAfter`.